test/component/uk/gov/hmrc/exports/ExportsSubmissionReceivedSpec.scala

```diff
@@ -17,240 +17,289 @@
-//
-//      And("the request was authorised with AuthService")
-//      eventually(verifyAuthServiceCalledForNonCsp())
-//
-//      And("the submission repository is not called")
-//      eventually(verifySubmissionRepositoryWasNotCalled())
-//
-//      And("the Declarations API Service is not called")
-//      eventually(verifyDecServiceWasNotCalled())
-//    }
-
-//    def testScenario(
-//      primeDecApiStubToReturnStatus: Int,
-//      submissionRepoMockedResult: Boolean,
-//      submissionRepoIsCalled: Boolean,
-//      expectedResponseStatus: Int,
-//      expectedResponseBody: String
-//    ): Unit = {
-//
-//      startSubmissionService(primeDecApiStubToReturnStatus)
-//      val request: FakeRequest[AnyContentAsXml] = ValidSubmissionRequest
-//
-//      Given("user is authorised")
-//      authServiceAuthorizesWithEoriAndNoRetrievals()
-//
-//      When("a POST request with data is sent to the API")
-//      val result: Future[Result] = route(app = app, request).value
-//
-//      And("submission should be handled")
-//      withSubmissionSuccess()
-//
-//      Then(s"a response with a $expectedResponseStatus status is received")
-//      status(result) shouldBe expectedResponseStatus
-//
-//      And(s"the response body is $expectedResponseBody")
-//      contentAsString(result) shouldBe expectedResponseBody
-//
-//      And("the Declarations API Service is called correctly")
-//      eventually(
-//        verifyDecServiceWasCalledCorrectly(
-//          requestBody = expectedSubmissionRequestPayload(declarantLrnValue),
-//          expectedEori = declarantEoriValue,
-//          expectedApiVersion = CustomsDeclarationsAPIConfig.apiVersion
-//        )
-//      )
-//
-//      if (submissionRepoIsCalled) {
-//        And("the submission repository is called correctly")
-//        eventually(verifySubmissionRepositoryIsCorrectlyCalled(declarantEoriValue))
-//      } else {
-//        And("the submission repository is not called")
-//        verifySubmissionRepositoryWasNotCalled()
-//      }
-//
-//      And("the request was authorised with AuthService")
-//      eventually(verifyAuthServiceCalledForNonCsp())
-//
-//      // TODO: do we need to test Audit service interaction? if so do it here
-//      // TODO: do we need to test NRS service interaction? if so do that here
-//    }
-//  }
+    scenario("an authorised user successfully submits a customs declaration") {
+      withSubmissionSuccess()
+      withNotificationRepositorySuccess()
+      withDeclarationRepositorySuccess()
+
+      startSubmissionService(ACCEPTED)
+      val request: FakeRequest[AnyContentAsJson] = ValidSubmissionRequest
+
+      Given("user is authorised")
+      authServiceAuthorizesWithEoriAndNoRetrievals()
+
+      When("a POST request with data is sent to the API")
+      val result = route(app = app, request).value
+
+      Then(s"a response with a $CREATED status is received")
```

Comment:
Is there any reason to use `$CREATED` instead just a string? Without this you don't need to use `s`.

Reply:
This has been addressed.